src/random.cpp

```diff
@@ -175,7 +175,20 @@ static void RandAddSeedPerfmon()
 #endif
 }
 
-#ifndef WIN32
+// URANDOM_FALLBACK should be defined if GetDevURandom(...) is used as a
+// fallback in GetOSRand(...).
+#if defined(WIN32)
```

Comment:
This is correct but I... think this is somewhat overdoing it, just to avoid a warning.

Reply:
This code just replicates the already existing `#ifdef` forest (implicit in the code below), but if we can avoid the warning in a smarter way I'm all ears. What method do you suggest using for avoiding the warning? :-)

Reply:
One option would be to forward-declare the method here and define it below based on a define that occurs inside the relevant #ifs.